[list partition] error when delete data

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
````sql
CREATE TABLE t1 (
                          dt date not null,
                          num int,
                          city varchar(20),
                          name varchar(20)
                        ) ENGINE=OLAP
                        DUPLICATE KEY(dt)
                        COMMENT "OLAP"
                        PARTITION BY LIST (dt) (
                           PARTITION p1 VALUES IN (("2022-04-01")),
                           PARTITION p2 VALUES IN (("2022-04-02")),
                           PARTITION p3 VALUES IN (("2022-04-03"))
                        )
                        DISTRIBUTED BY HASH(dt) BUCKETS 3
                        PROPERTIES (
                        "replication_num" = "1"
                        );
INSERT INTO t1 VALUES ('2022-04-01', 1, null, 'jack'),
                                       ('2022-04-02', 1, null, 'jack'),
                                       ('2022-04-03', 1, null, 'jack');
DELETE FROM t1 WHERE num=1;
````
### Expected behavior (Required)
`delete succeed`
### Real behavior (Required)
`ERROR 1064 (HY000): Index: 0, Size: 0`
```
2023-02-16 17:26:23,762 WARN (starrocks-mysql-nio-pool-18|715) [DeleteHandler.process():182] error occurred during delete process
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:657) ~[?:1.8.0_252]
	at java.util.ArrayList.get(ArrayList.java:433) ~[?:1.8.0_252]
	at com.starrocks.load.DeleteHandler.checkDelete(DeleteHandler.java:535) ~[starrocks-fe.jar:?]
	at com.starrocks.load.DeleteHandler.createJob(DeleteHandler.java:240) ~[starrocks-fe.jar:?]
	at com.starrocks.load.DeleteHandler.process(DeleteHandler.java:175) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.StmtExecutor.handleDMLStmt(StmtExecutor.java:1239) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:488) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:342) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:459) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:723) ~[starrocks-fe.jar:?]
	at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:68) ~[starrocks-fe.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
2023-02-16 17:26:23,762 WARN (starrocks-mysql-nio-pool-18|715) [StmtExecutor.execute():493] DML statement(DELETE FROM t1 WHERE num=1) process failed.
com.starrocks.common.DdlException: Index: 0, Size: 0
	at com.starrocks.load.DeleteHandler.process(DeleteHandler.java:188) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.StmtExecutor.handleDMLStmt(StmtExecutor.java:1239) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:488) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:342) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:459) ~[starrocks-fe.jar:?]
	at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:723) ~[starrocks-fe.jar:?]
	at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:68) ~[starrocks-fe.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:657) ~[?:1.8.0_252]
	at java.util.ArrayList.get(ArrayList.java:433) ~[?:1.8.0_252]
	at com.starrocks.load.DeleteHandler.checkDelete(DeleteHandler.java:535) ~[starrocks-fe.jar:?]
	at com.starrocks.load.DeleteHandler.createJob(DeleteHandler.java:240) ~[starrocks-fe.jar:?]
	at com.starrocks.load.DeleteHandler.process(DeleteHandler.java:175) ~[starrocks-fe.jar:?]
	... 9 more
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
`MAIN-RELEASE-f256177`

--- a/fe/fe-core/src/main/java/com/starrocks/load/DeleteHandler.java
+++ b/fe/fe-core/src/main/java/com/starrocks/load/DeleteHandler.java
@@ -54,6 +54,7 @@ import com.starrocks.analysis.SlotRef;
 import com.starrocks.catalog.Column;
 import com.starrocks.catalog.Database;
 import com.starrocks.catalog.KeysType;
+import com.starrocks.catalog.ListPartitionInfo;
 import com.starrocks.catalog.MaterializedIndex;
 import com.starrocks.catalog.MaterializedIndexMeta;
 import com.starrocks.catalog.OlapTable;
@@ -211,16 +212,25 @@ public class DeleteHandler implements Writable {
         Preconditions.checkState(partitionNames != null);
         boolean noPartitionSpecified = partitionNames.isEmpty();
         if (noPartitionSpecified) {
-            if (olapTable.getPartitionInfo().getType() == PartitionType.RANGE) {
+            PartitionInfo partitionInfo = olapTable.getPartitionInfo();
+            if (partitionInfo.isRangePartition()) {
                 partitionNames = extractPartitionNamesByCondition(stmt, olapTable);
                 if (partitionNames.isEmpty()) {
                     LOG.info("The delete statement [{}] prunes all partitions",
                             stmt.getOrigStmt().originStmt);
                     return null;
                 }
-            } else if (olapTable.getPartitionInfo().getType() == PartitionType.UNPARTITIONED) {
+            } else if (partitionInfo.getType() == PartitionType.UNPARTITIONED) {
                 // this is a unpartitioned table, use table name as partition name
                 partitionNames.add(olapTable.getName());
+            } else if (partitionInfo.getType() == PartitionType.LIST) {
+                // TODO: support list partition prune
+                ListPartitionInfo listPartitionInfo = (ListPartitionInfo) partitionInfo;
+                List<Long> partitionIds = listPartitionInfo.getPartitionIds(false);
+                for (Long partitionId : partitionIds) {
+                    Partition partition = olapTable.getPartition(partitionId);
+                    partitionNames.add(partition.getName());
+                }
             }
         }
 

